src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -139,16 +139,21 @@ bool layout_optimizer::can_fuse_reorder(program_node& prev, program_node& next,
     auto next_dt = next.get_output_layout().data_type;
     auto use_onednn_impls = has_all_enabled_onednn_impls_optimization_attribute();
 
-    if (prev.is_dynamic() || next.is_dynamic())
-        return false;
-
     auto is_input_idx = [&](size_t idx) -> bool {
         if (&next.get_dependency(idx) == &prev)
             return true;
         if (next.get_dependency(idx).is_type<reorder>() && &next.get_dependency(idx).get_dependency(0) == &prev)
             return true;
         return false;
     };
+    auto get_channel_count = [](const layout& layout) -> int64_t {
+        return layout.get_partial_shape()[1].is_static() ? layout.get_partial_shape()[1].get_length() : -1;
+    };
```

Comment:
what about making a helper function to collect channel count based on weight? Here's code from Wilson that is in review as well:

https://github.com/openvinotoolkit/openvino/pull/31990/files#diff-41a32bc38b6fa9c7db5e656050c380c993d424df879bb6a1e2627e7008a7704bR972

Reply:
Applied